toml/decoder.py

```diff
@@ -957,3 +964,54 @@ def load_array(self, a):
                     atype = ntype
                 retval.append(nval)
         return retval
+
+    def handle_inline_comment(self, line, currentlevel):
+        return line
+
+    def handle_comment(self, currentlevel, comment_no, recent, line):
+        return comment_no
+
+    def comment_lint(self, sl, i, beginline):
+        j = i
+        try:
+            while sl[j] != '\n':
+                sl[j] = ' '
+                j += 1
+        except IndexError:
+            raise IndexError()
+        return beginline
+
+
+class TomlPreserveCommentDecoder(TomlDecoder):
+
+    def __init__(self, _dict=OrderedDict):
+        super(TomlPreserveCommentDecoder, self).__init__(_dict)
+
+    def handle_inline_comment(self, line, currentlevel):
+        pos = line.find('#')
+        k = pos
+        comment = ""
+        try:
+            while line[k] != '\n':
+                comment += line[k]
+                k += 1
+        except IndexError:
+            pass
+        pair = line.split('=')
+        currentlevel[pair[0].strip() + '-' + 'comment' + '_inline'] = comment
```

Comment:
Doesn't this interfere with keys that end in `-comment_inline` ?
Not likely, but prone to weirdness. I think a custom dictionary type that keeps the comments outside of the normal key-space of the dictionary would eliminate that class of bugs.

Reply:
```python
class CommentKey:
    def __init__(self, key, val):
        self.key = key
        self.val = val

    def __hash__(self):
        return hash(str(self))

    def __str__(self):
        return '{}, {}'.format(self.key, self.val)

    def __eq__(self, other):
        return str(self) == other
```
Something like this ?